### PR TITLE
Add linux tests, new centos7 image

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Notes
- like #18, adds gtk3
  - only on linux, however
- motivated by https://github.com/conda-forge/geckodriver-feedstock/pull/11
- adds `yum_requirements.txt`, per https://github.com/conda-forge/gtk3-feedstock/issues/16
- breaks up test into a few smaller pieces (could still be better)
- tries to screenshot
- tries pytest-azurepipelines so we might get some history out of it

### CODE REVIEW

**Comment by bollwyvl** [comment]
Linux got [further](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=191127&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=1217):

```
Checking geckodriver.log...

$PREFIX/bin/FirefoxApp/firefox: /lib64/libc.so.6: version `GLIBC_2.17' not found (required by $PREFIX/bin/FirefoxApp/firefox)
$PREFIX/bin/FirefoxApp/firefox: /lib64/libc.so.6: version `GLIBC_2.14' not found (required by $PREFIX/bin/FirefoxApp/firefox)


```

**Comment by bollwyvl** [comment]
So, looking at the hits for those errors, this isn't going to get fixed. Have restored the linux skips. We might just want to hold onto this for now, as the tests are more expressive, but it doesn't _really_ change the overall packageability, due to the `yum_requirements.txt`, and I'm still not ready to invest in doing a full source build, as this package pretty much works on modern linux systems.

**Comment by bollwyvl** [comment]
Hooray/boo green.

**Comment by bollwyvl** [comment]
Ah, good news: we can pull the Centos7 docker image, now, which might have a new-enough glibc... also i (re) learned that `bin/firefox` is not a firefox executable, but rather a bash script... geckodriver `0.27.0` is displeased by this. push incoming, once verified locally...

**Comment by bollwyvl** [comment]
Progress:
```
XPCOMGlueLoad error for file $PREFIX/bin/FirefoxApp/libmozgtk.so:
libgtk-3.so.0: cannot open shared object file: No such file or directory
```

**Comment by birdsarah** [comment]
Was def hoping we could depend on gtk so that i don't need extra bash script pieces when running crawlers, but i don't think it's worth adding gtk3 just to pass our test suite.

**Comment by bollwyvl** [comment]
And yet it passes!

It has _incidentally_ been working, but there has always been this nagging sense in the back of my mind that I won't know until some downstream test fails when we ship something. I also rely on these things _working_, and usually browser testing comes at the long end of a build, so would be happier to know that it has at least been tested _once_ so that I can focus on... browser flake, mostly, I guess.
 
Anyhow: I feel like it probably is worth merging this so that we have a bit more assurance when we ship. Barring any concerns (the yum thing takes about more 20s, I guess) I'm :+1: to merging this.

File: .scripts/build_steps.sh [hunk 1]
```diff
@@ -26,16 +26,23 @@ setup_conda_rc "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
...
 make_build_number "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
 
 conda build "${RECIPE_ROOT}" -m "${CI_SUPPORT}/${CONFIG}.yaml" \
     --suppress-variables \
     --clobber-file "${CI_SUPPORT}/clobber_${CONFIG}.yaml"
-validate_recipe_outputs "${FEEDSTOCK_NAME}"
 
 if [[ "${UPLOAD_PACKAGES}" != "False" ]]; then
-    upload_package --validate --feedstock-name="${FEEDSTOCK_NAME}"  "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
+    upload_package  "${FEEDSTOCK_ROOT}" "${RECIPE_ROOT}" "${CONFIG_FILE}"
```
Comment (birdsarah) [inline .scripts/build_steps.sh:45]: How come we've lost `validate_recipe_outputs` and `--validate`, does that happen somewhere else?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -17,19 +17,30 @@ source:
...
     - selenium
     - geckodriver
+    - pytest-azurepipelines
+    - python >=3.6
+  commands:
+    - copy {{ RECIPE_DIR }}\test_selenium.py .  # [win]
+    - copy {{ RECIPE_DIR }}\setup.cfg .  # [win]
+    - cp {{ RECIPE_DIR }}/test_selenium.py {{ RECIPE_DIR }}/setup.cfg .  # [not win]
+    # TODO: may be needed
+    # - set -m && % xattr -r -d com.apple.quarantine $PREFIX/bin/geckodriver  # [osx]
```
Comment (birdsarah) [inline recipe/meta.yaml:41]: Shall we delete this from this PR to keep things clean?

File: recipe/setup.cfg [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[tool:pytest]
+junit_family = xunit2
```
Comment (birdsarah) [inline recipe/setup.cfg:2]: Just for my information. Is this so that azure_pipelines can consume the output?

File: recipe/test_selenium.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            service_args=["--log", "trace"],
+            **binary_paths,
+        )
+        driver.get("about:license")
+
+        if driver.page_source:
+            html_log.write_text(driver.page_source)
+            assert "Mozilla Public License 2.0" in driver.page_source, \
+                "couldn't even load the license page"
+            driver.save_screenshot(str(license_png))
```
Comment (birdsarah) [inline recipe/test_selenium.py:67]: I'm on the fence about this. Am I right in thinking we wouldn't be able to get the saved png out of the azure box to view it? If yes then we're just testing the save_screenshot capability of geckodriver, which feels out of scope.

But if you feel strongly about it that's fine.

File: .scripts/build_steps.sh (hunk 1 above)
Comment (birdsarah) [reply .scripts/build_steps.sh:45]: (same line has been changed in an osx script too)

perhaps this is just a a conda-smithy thing, but just wanted to check.

Comment (bollwyvl) [reply .scripts/build_steps.sh:45]: yeah, it's just smithy. not my explicit doing.

File: recipe/meta.yaml (hunk 2 above)
Comment (bollwyvl) [reply recipe/meta.yaml:41]: nah, if we even think it's going to be needed in the future, i think we should keep it. Indeed, i should probably add a link to the canonical page.

File: recipe/setup.cfg (hunk 3 above)
Comment (bollwyvl) [reply recipe/setup.cfg:2]: Right. And it seems to be working, which is awesome! Even though the build logs roll off after a few months, the test results persist (for at least a year)... knowing which previous builds were from which PRs might help us avoid future rework.

File: recipe/test_selenium.py (hunk 4 above)
Comment (bollwyvl) [reply recipe/test_selenium.py:67]: screenshots are my bread and butter. also, you _can_ extract them back out from earlier in the log, as they get captured as base64. maybe there's a way to add them as attachments to the xunit... will look into it...

**Comment by bollwyvl** [comment]
>  load about:support and check that the version number that this package is building is in the page. 

Yes, will do this in addition to the license page... who knows when `MPL-3` will come out :rofl:

File: recipe/test_selenium.py (hunk 4 above)
Comment (birdsarah) [reply recipe/test_selenium.py:67]: Here's how screenshot embedding is done with pytest-html (https://github.com/conda-forge/firefox-feedstock/pull/26#discussion_r463805394), don't know about xunit.

Comment (birdsarah) [reply recipe/test_selenium.py:67]: interesting, there's some kind of attachments option: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=191234&view=ms.vss-test-web.build-test-results-tab&runId=1146678&resultId=100000&paneView=attachments

Comment (bollwyvl) [reply recipe/test_selenium.py:67]: ah, it's just a very thin wrapper around `pytest-nunit`... will do a staged-recipes for that, but keep the current behavior for now.

Comment (bollwyvl) [reply recipe/test_selenium.py]: https://github.com/conda-forge/staged-recipes/pull/12273

File: recipe/meta.yaml (hunk 2 above)
Comment (bollwyvl) [reply recipe/meta.yaml:42]: aaf61f5

**Comment by birdsarah** [comment]
looks good to me. merge when you're ready.

**Comment by birdsarah** [comment]
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=191553&view=results

do you know why sometimes there's a tests tab, and sometimes there isn't?

**Comment by bollwyvl** [comment]
> sometimes there's a tests tab, and sometimes there isn't?

:shrug: i guess if it breaks we will probably have to fight with it enough times that the magic condition will be met? Also, it seems sometimes that following the links from here to there vs going from the top end is neccessary to get there? there's also the "analytics" tab from the top-level that shows all the runs, which can 

I've usually done some of the heavyweight azure stuff on other things, but i'll take the simplicity of one package, no-config here... but if it consistently isn't showing, we'll have to investigate further...
